src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java

```diff
@@ -53,4 +60,45 @@ public ResponseEntity<GetGrantSchemeWithApplicationAndAdverts> getGrantSchemeByI
                 .build();
         return ResponseEntity.ok(getGrantSchemeWithApplicationAndAdverts);
     }
+    @GetMapping("/{grantSchemeId}/hasInternalApplication")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Check if grant scheme is for internal or external application",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = Boolean.class))),
+            @ApiResponse(responseCode = "404", description = "No Grant Scheme found",
+                    content = @Content(mediaType = "application/json")),
+    })
+    public ResponseEntity<SchemeMandatoryQuestionApplicationFormInfosDto> schemeHasInternalApplication(@PathVariable Integer grantSchemeId){
+        log.info("Checking if scheme with id {} has internal application form and if it is published",grantSchemeId);
+
+        log.info("Getting scheme with id {}", grantSchemeId);
```

Comment:
There are quite a lot of logs here, not sure we want to flood the logs like this.

Reply:
shall i move some of them to a different level (debug) or just remove them?